test/grouping.jl

```diff
@@ -2691,4 +2691,122 @@ end
+        else
+            @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+        end
+    end
+
+    for fun in (sum∘skipmissing, mean∘skipmissing),
+        col in ([1:3, 4:6, 7:9], [1:3, 4:6, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    # workaround https://github.com/JuliaLang/julia/issues/36979
+    for fun in (var∘skipmissing, std∘skipmissing),
+        col in ([1:3, 4:6, 7:9], [1:3, 4:6, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test_throws MethodError combine(gdf, :x => fun => :y)
+        @test_throws MethodError combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    for fun in (maximum, minimum, maximum∘skipmissing, minimum∘skipmissing,
+                first, last, length, first∘skipmissing, last∘skipmissing),
+        col in ([1:3, 4:6, 7:9], [1:3, 4:6, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        if fun isa typeof(last∘skipmissing)
+            @test_throws MethodError combine(gdf, :x => fun => :y)
+            @test_throws MethodError combine(gdf, :x => (x -> fun(x)) => :y)
+        else
+            @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+        end
+    end
+
+    for fun in (prod, prod∘skipmissing),
+        col in ([1:3, 4:6, 7:9], [1:3, 4:6, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test_throws MethodError combine(gdf, :x => fun => :y)
+        @test_throws MethodError combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    for fun in (sum, prod, mean, var, std, sum∘skipmissing, prod∘skipmissing,
+                mean∘skipmissing, var∘skipmissing, std∘skipmissing,
+                maximum, minimum, maximum∘skipmissing, minimum∘skipmissing,
+                first, last, length, first∘skipmissing, last∘skipmissing),
+        col in ([ones(2,2), zeros(2,2), ones(2,2)], [ones(2,2), zeros(2,2), missing],
+                [DataFrame(ones(2,2)), DataFrame(zeros(2,2)), DataFrame(ones(2,2))],
+                [DataFrame(ones(2,2)), DataFrame(zeros(2,2)), ones(2,2)],
+                [DataFrame(ones(2,2)), DataFrame(zeros(2,2)), missing],
+                [(a=1, b=2), (a=3, b=4), (a=5, b=6)], [(a=1, b=2), (a=3, b=4), missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        if fun isa typeof(length)
+            @test combine(gdf, :x => fun => :y) ≅ DataFrame(g=1, y=3)
+            @test combine(gdf, :x => (x -> fun(x)) => :y) ≅ DataFrame(g=1, y=3)
+        elseif (fun isa typeof(last) && ismissing(last(col))) ||
+               (fun isa Union{typeof(maximum), typeof(minimum)} && col ≅ [(a=1, b=2), (a=3, b=4), missing])
+            # this case is a hard problem what to do and probably we have to leave it as is
```

Comment:
I must admit I don't follow... :-p

Reply:
I want to highlight that we do `collect` by value not by collection type, so in some cases, although a collection in general would not get accepted (it contains `NamedTuple`s on which we throw an error), it happens to be accepted because the transformation function returns a value that is accepted: `missing`.

Normally we would error like this:
```
julia> df = DataFrame(g=[1,1,2,2], x=[(a=1,b=2),(a=1,b=2), (a=1,b=2), missing])
4×2 DataFrame
│ Row │ g     │ x              │
│     │ Int64 │ NamedTuple…?   │
├─────┼───────┼────────────────┤
│ 1   │ 1     │ (a = 1, b = 2) │
│ 2   │ 1     │ (a = 1, b = 2) │
│ 3   │ 2     │ (a = 1, b = 2) │
│ 4   │ 2     │ missing        │

julia> gdf = groupby(df, :g);

julia> combine(x -> last(x.x), gdf)
ERROR: ArgumentError: function must return only single-column values, or only multiple-column values
```
but in this special case we do not.